valhalla/thor/route_matcher.h

```diff
@@ -39,8 +39,7 @@ class RouteMatcher {
                        const sif::TravelMode& mode,
                        baldr::GraphReader& reader,
                        const std::vector<meili::Measurement>& shape,
-                       const bool use_timestamps,
-                       const google::protobuf::RepeatedPtrField<valhalla::Location>& correlated,
+                       valhalla::Options& options,
```

Comment:
it is somehow against intuitive to pass options as & then modify it. Maybe need another name or way.

Reply:
the problem is that we need to set time information on the locations. this is related to the TODOs i added in the functions that call these map matching FromPath functions. we could move all of the date setting there but that is a bit hairy because we are supporting timestamps as a date which means we need to get a node to turn it into tz and then turn the epoch into date string. the whole thing could probably be refactored to be easier to understand.

as it stands though this is inline with the request flow throughout the process. essentially as we fulfill the request we fill in bits of info along the way, which all hangs off of the `api` object, which owns this `options` object.